src/bot/commands/user_commands.py

```diff
@@ -59,15 +59,24 @@ def run_get_player_recents_command(update, context):
     )
     limit = constants.QUERY_PARAMETERS.RESPONSE_LIMIT.value
 
+    hero_name_parts = context.args
+    parts_to_remove = []
+
     for arg in context.args:
         user = user_services.lookup_user_by_telegram_handle(arg)
         if user:
             registered_user = user
+            parts_to_remove.append(arg)
+
         try:
             limit = int(arg)
+            parts_to_remove.append(arg)
```

Comment:
This can be cleaned up by using the `list.pop()` method and `list.index()` method. Pop will return the element at the rear of the list, or takes an optional argument with the position of the element to remove. This would look like the following: `list.pop(list.index(arg))`. Would allow us to capture the return value immediately if we're able to convert it to an int (which means its our limit) or if a user is returned from the service. Would mean the secondary loop on line 77 would no longer be necessary.

Reply:
The iterator always goes to the n+1th item in the list and doesn't keep track of whether the current item changes positions. Removing things from a list while iterating over it causes items to get skipped when their position changes. This was the cleanest way I could make sure to catch all inputs.